### ISSUE TITLE
Bug: failing tests on Haiku

### ISSUE THREAD

**Issue by Begasus** [assimp/assimp#5538]
**Describe the bug**
running the test suite leads to failures and even one is crashing

**To Reproduce**
1. Build the package
2. Check the testsuite while running the binary "unit"

**Expected behavior**
A full (or at least not crashing) testsuite (version 5.3.1 passed all tests)

**Platform (please complete the following information):**
 - OS: [Haiku R1B4]
 - Browser [Falkon]
 - Version [5.83.04]

**Additional context**
Adding Terminal output and a bug report
[assimp-5.4.0_tests.txt](https://github.com/assimp/assimp/files/14916650/assimp-5.4.0_tests.txt)
[unit-15669-debug-09-04-2024-06-00-42.report.txt](https://github.com/assimp/assimp/files/14916654/unit-15669-debug-09-04-2024-06-00-42.report.txt)

**Comment by Begasus** [comment]
This is quite a big file to read through, hope you can find something usefull in it, I used `build/bin/unit > unit.debug.txt` because I couldn't grab the full log in Terminal.
What left shown in Terminal was the following:
```
Testing ...
nullptr returned by creating DDLNode.
Warning: No cameras found in scene file.
0 / 19 meshes instanced.
textures/wal67ar_small.jpg: must convert this texture to PNG.
textures/wal67ar_small.png: unable to load texture and check for alpha mask in texture. Geometry will not be alpha masked with this texture.
textures/wal69ar_small.jpg: must convert this texture to PNG.
textures/wal69ar_small.png: unable to load texture and check for alpha mask in texture. Geometry will not be alpha masked with this texture.
textures/SpiderTex.jpg: must convert this texture to PNG.
textures/SpiderTex.png: unable to load texture and check for alpha mask in texture. Geometry will not be alpha masked with this texture.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.png: unable to load texture and check for alpha mask in texture. Geometry will not be alpha masked with this texture.
textures/engineflare1.jpg: must convert this texture to PNG.
textures/engineflare1.png: unable to load texture and check for alpha mask in texture. Geometry will not be alpha masked with this texture.
textures/wal67ar_small.jpg: must convert this texture to PNG.
textures/wal69ar_small.jpg: must convert this texture to PNG.
textures/SpiderTex.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/engineflare1.jpg: must convert this texture to PNG.
textures/SpiderTex.jpg: must convert this texture to PNG.
textures/wal67ar_small.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/wal67ar_small.jpg: must convert this texture to PNG.
textures/wal67ar_small.jpg: must convert this texture to PNG.
textures/wal67ar_small.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/drkwood2.jpg: must convert this texture to PNG.
textures/engineflare1.jpg: must convert this texture to PNG.
textures/engineflare1.jpg: must convert this texture to PNG.
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 1.
cleaning 'chroot/boot' folder
keeping chroot folder /Opslag/haikuports/media-libs/assimp/work-5.4.0 intact for inspection
Error: Test has failed - stopping.
```

[unit.debug.txt](https://github.com/assimp/assimp/files/14948208/unit.debug.txt)

**Comment by kimkulling** [comment]
The revision will be zero in all releases, so the test is only valid for development versions. I will prepare a fix for that and release 5.4.1 .

See https://github.com/assimp/assimp/issues/5564 for tracing the issue.
